tests/testthat/test-summary-log.R

```diff
@@ -217,3 +218,47 @@ test_that("summary_log() works with filtering parameter string [BBR-SMLG-011]",
             expect_equal(summary_log(MODEL_DIR, .include = c(1:2, "Parent")) %>% nrow(), 3)
 
           })
+
+test_that("summary_log() and add_summary() calculate aic and bic by default", {
+  clean_test_enviroment(create_all_models)
+  copy_all_output_dirs()
+  sum_df <- summary_log(MODEL_DIR)
+  expect_true(all(c(AIC_COL, BIC_COL) %in% names(sum_df)))
+
+  # Can be turned off (e.g., for bootstrap runs)
+  sum_df <- summary_log(MODEL_DIR, calc_aic_bic = FALSE)
+  expect_false(all(c(AIC_COL, BIC_COL) %in% names(sum_df)))
+
+  # add_aic_bic appends AIC and BIC columns
+  sum_df <- sum_df %>% add_aic_bic()
+  expect_true(all(c(AIC_COL, BIC_COL) %in% names(sum_df)))
+
+  # Same behavior with add_summary
+  log_df <- run_log(MODEL_DIR) %>% add_summary()
+  expect_true(all(c(AIC_COL, BIC_COL) %in% names(log_df)))
+
+  log_df <- run_log(MODEL_DIR) %>% add_summary(calc_aic_bic = FALSE)
+  expect_false(all(c(AIC_COL, BIC_COL) %in% names(log_df)))
+})
+
+test_that("add_dofv() works correctly", {
```

Comment:
@barrettk:

> I did have some concerns about the tests, specifically for calculating `dofv`. Given that we only have one model in the repo, we would only ever get a difference of `0`, without adjusting the model and re-running it.

Thanks for highlighting this. I agree with that concern.

Rather than rerunning, you can probably get by with some custom test setup that adjusts the `OBJV` line for the test model lst's so that they have different and known values, letting you check for an expected value.

In addition to tweaking the OBJV values and testing for expected values, it'd be good to assert that calls with a run log and summary log for the same set of models give the same dofv values.

Reply:
I said:

>  some custom test setup that adjusts the OBJV line
> [...]
> it'd be good to assert that calls with a run log and summary log for the same set of models give the same dofv values.

Both these are done with 2cd0480b.